### ISSUE TITLE
Android App does not connect to server

### ISSUE THREAD

**Issue by Florikoeln** [stefan-niedermann/nextcloud-deck#1080]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**

Deck (and also the other nextcloud apps installed from fdroid) cannot connect to my private hosted nextcloud. Connection via browser on android device is possible. Issue comes up on WiFi and LTE. Restart of app, deletion of cashe and restarting device does not help.

1. Open App
2. See error

**Expected behavior**

Deck connects to server

**Screenshots**

Output by Deck:

App Version: 1.19.3
App Version Code: 1019003
Server App Version: 1.4.3
App Flavor: fdroid

Files App Version Code: 30160190

---

OS Version: 4.4.194-perf+(00WW_2_400_SP04)
OS API Level: 29
Device: SLD_sprout
Manufacturer: HMD Global
Model (and Product): Nokia 6.2 (Starlord_00EEA)

---

com.nextcloud.android.sso.exceptions.UnknownErrorException: failed to connect to (URL) (port 443) from /2a0a:a546:e4c8:0:8463:73ce:bd4e:39d4 (port 39908) after 60000ms: isConnected failed: ECONNREFUSED (Connection refused)
	at com.nextcloud.android.sso.api.AidlNetworkRequest.performNetworkRequestV2(AidlNetworkRequest.java:172)
	at com.nextcloud.android.sso.api.NextcloudAPI.performNetworkRequestV2(NextcloudAPI.java:171)
	at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$NextcloudAPI(NextcloudAPI.java:110)
	at com.nextcloud.android.sso.api.NextcloudAPI$$ExternalSyntheticLambda0.subscribe(Unknown Source:6)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

**Comment by desperateCoder** [comment]
Hi @Florikoeln, thanks for your report!

That's wild! Never saw anything like this before in our app. I highly suspect an configuration error here, but yeah, let's find out.

Great catch to omit the url in your report. What happens, if you just enter it into a browser? I mean the **full** url provided in the error message. I suspect the server side just refuses the connection.

Does the Nextcloud **Files** app work properly? Can you still upload files?

**Comment by Florikoeln** [comment]
Hi @desperateCoder, thanks for the quick answer.

I have tried:

1. connect to URL (w/o https) in browser (Phone, same WIFI as Nextcloud) --> nextcloud login Window
2. connect to URL (w/o https) in browser (Phone, LTE) --> netcloud login Window
3. connect to URL (with "https://") in browser (Phone, same WIFI as Nextcloud) --> nextcloud login Window
4. connect to URL (with "https://") in browser (Phone, LTE) --> nextcloud login Window
5. connect to IPv6 IP address provided in error message: browser misinterprets as search request

In case 1-4 I can login into Nextcloud and upload files. 

I have also installed "NC Passwords", "Nextcloud", "Notizen" (all are the current fdroid versions), none of these can connect. 

Maybe I should note that 
- the certificate (Let's encrypt) was recently renewed, this is why I tried to clear the app's cache,
- my nextcloud installation was installed by snap (nextcloud.occ status:  - installed: true - version: 21.0.3.1 - versionstring: 21.0.3 edition: )

**Comment by desperateCoder** [comment]
> Edit: connect to IPv6 IP address provided in error message (https://[2001:...]: connection refused (ERR_CONNECTION_REFUSED)

I think that's the point. Noone knows what is behind the IP right now. you could try to resolve the domain to the current IP, I guess it will be a different one now. The primary question is: how the hell did the Files app think it's a good idea to use the IP?! did you use a proxy or a VPN at setup time? Maybe something did automatically translate your domain to an IP?

Can you just use the actual domain URL (not the IP one) to connect your files app to your server? Once the files app works, Deck will also start working, since we use the network stack from the files app. It does the actual requests for Deck, that's why you need the Files app to be installed to use Deck.

> the certificate (Let's encrypt) was recently renewed, this is why I tried to clear the app's cache,

From my understanding this shouldn't affect anything here. That's totally normal and happens all the time. 

> I tried the (appstore) app on an IOs device, works without any problem (same account) on both, Wifi as well as LTE.

I guess it doesn't rely on the Files app. And it most likely doesn't use the IP, but the Domain. As mentioned above, just try out to set it up with the domain and It should work. 

Feedback would be appreciated, although Deck hasn't any errors here from what I can see.

**Comment by Florikoeln** [comment]
I finally solved the issue:

1. Loggin into myfritz.net
2. Checked the full URL of the server (is in my case <computername>.<numbers and letters>.myfritz.net
3. Added the URL to "trusted domains" in config.php
3. Used this URL in the android files app --> login possible
4. removed account <numbers and letters>.myfritz.net in Deck 
5. added new <computername>.<numbers and letters>.myfritz.net to Deck --> OK

Thanks for the support.
